src/components/Admin/AddQuestion.js

```diff
@@ -0,0 +1,259 @@
+    description: "",
+    loading: true,
+    validated: false,
+    sending: false,
+    submitted: false,
+    error: null,
+  };
+
+  componentDidMount() {
+    if (this.props.firebase && !this._initFirebase) this.loadData();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.firebase && !this._initFirebase) this.loadData();
+    if (typeof window !== "undefined") {
+      import("bs-custom-file-input").then((bsCustomFileInput) => {
+        bsCustomFileInput.init();
+      });
+    }
+  }
+
+  loadData = async () => {
+    this._initFirebase = true;
+
+    this.setState({
+      loading: false,
+    });
+  };
+
+  onSubmit = (event) => {
+    this.setState({ sending: true });
+
+    const { answer, description, name, file, fileExtension } = this.state;
+
+    if (answer === "" || description === "" || name === "") {
+      this.setState({ validated: false });
+    } else {
+      this.setState({ validated: true });
+      var im = this.props.uid + "." + fileExtension;
+      if (file === null) {
+        im = "";
+      }
+
+      const data = {
+        answer: answer,
+        description: description,
+        name: name,
+        image: im,
+        scores: {},
+      };
+
+      if (file !== null) {
+        var uploadTask = this.props.firebase.file("Questions", im).put(file);
+
+        uploadTask.on(
+          "state_changed",
+          function (snapshot) {
+            var progress =
+              (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
+            console.log("Upload is " + progress + "% done");
+          },
+          function (error) {
+            console.log(error);
+          },
+          function () {
+            console.log("Upload Successful!");
+          }
+        );
+      }
+
+      this.props.firebase
+        .questions()
+        .add(data)
+        .then(() => {
+          console.log("Added Question");
+          this.setState(
+            {
+              submitted: true,
+            },
+            () => {
+              this.props.updateFunc();
+            }
+          );
+        })
+        .catch((error) => {
+          this.setState({ error });
+        });
+    }
+
+    this.setState({ sending: false });
+    event.preventDefault();
+  };
+
+  onChange = (event) => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  onFileChange = (event) => {
+    var f = event.target.files[0];
+    console.log(f);
+    if (f.type !== "image/jpeg" && f.type !== "image/png") {
+      console.log("Invalid file type");
+      f = null;
+    } else {
+      var fileExtension = "jpg";
+      if (f.type.split("/")[1] === "png") fileExtension = "png";
+      this.setState({ fileExtension: fileExtension });
+    }
+
+    this.setState({ file: f });
+  };
+
+  render() {
+    const {
+      fileExtension,
+      file,
+      name,
+      answer,
+      description,
+      loading,
+      error,
+      submitted,
+      validated,
+      sending,
+    } = this.state;
```

Comment:
fileExtension, file, and submitted are all not being used.

Reply:
Resolved in commit 15bc7feade4e9e37cdcfb0f3f5de88ed6c4b5193